### PR TITLE
Detect Eclipse300 license problems

### LINKED ISSUE #8728: License errors from Eclipse300 will not trigger a rerun
ecl_run will for Eclipse100 detect license issues and then retry later, but the ecl_run wrapper is not able to do the same for Eclipse300 which has a different looking error message when a license error occurs:

```
 @--Message:The message service has been activated
 @--Message:Checking for licenses
 @--Message:Checking for licenses
 @--Message:Checking for licenses
 @--Error  
 @ ECLIPSE option not allowed in license                                   
 @ Please ask for a new license                                            
 @ Run stopping                                                            
            0 Mbytes of storage required   
```

This error message was obtained by perturbing the environment variable in `runeclipse` pointing to the license server and then running  `runeclipse -p e300 -i EIGHTCELLS.DATA` with runeclipse from subscript and EIGHTCELLS.DATA from res2df

### PR DESCRIPTION
**Pull request by berland**
**Issue**
Resolves lack of testing and resolves #8728

**Approach**
:brain:

- [x] PR title captures the intent of the changes, and is fitting for release notes.
- [x] Added appropriate release note label
- [x] Commit history is consistent and clean, in line with the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md).
- [x] Make sure unit tests pass locally after every commit (`git rebase -i main
      --exec 'pytest tests/unit_tests -n logical -m "not integration_test"'`)

## When applicable
- [ ] **When there are user facing changes**: Updated documentation
- [x] **New behavior or changes to existing untested code**: Ensured that unit tests are added (See [Ground Rules](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md#ground-rules)).
- [ ] **Large PR**: Prepare changes in small commits for more convenient review
- [x] **Bug fix**: Add regression test for the bug
- [ ] **Bug fix**: Create Backport PR to latest release

### CODE REVIEW

File: src/ert/resources/forward-models/res/script/ecl_run.py [hunk 1]
```diff
@@ -12,12 +12,17 @@
...
 from packaging import version
 
 
 class EclError(RuntimeError):
-    pass
+    def failed_due_to_license_problems(self) -> bool:
+        return (
+            "LICENSE ERROR" in self.args[0]
+            or "LICENSE FAILURE" in self.args[0]
+            or "not allowed in license"
```
Comment (berland) [inline src/ert/resources/forward-models/res/script/ecl_run.py:24]: aj, bug, missing  `in self.args[0]`

Comment (berland) [reply src/ert/resources/forward-models/res/script/ecl_run.py:24]: add test for happy path.

File: tests/unit_tests/resources/test_ecl_run_new_config.py [hunk 2]
```diff
@@ -267,3 +267,71 @@ def test_summary_block(source_root):
...
+ Bugs                   0
+ Final cpu       0.01 elapsed       0.02
+ Emergency stop called from routine   ZSTOPE"""
+
+    Path("FOO.PRT").write_text(prt_error + "\n" + eclend, encoding="utf-8")
+    Path("FOO.ECLEND").write_text(eclend, encoding="utf-8")
+    Path("FOO.DATA").write_text("", encoding="utf-8")
+
+    run = ecl_run.EclRun("FOO.DATA", "dummysimulatorobject")
+    try:
```
Comment (eivindjahren) [inline tests/unit_tests/resources/test_ecl_run_new_config.py:333]: Could use 
```python
with pytest.raises(ecl_run.EclError) as exc_info:
  run.assertECLEND()
assert exc_info.value.failed_due_to_license_problems()
```

**Review by eivindjahren — APPROVED** [review]
Small suggestion about using `pytest.raises` otherwise looks good :)

File: src/ert/resources/forward-models/res/script/ecl_run.py (hunk 1 above)
Comment (berland) [reply src/ert/resources/forward-models/res/script/ecl_run.py]: fixed

File: tests/unit_tests/resources/test_ecl_run_new_config.py (hunk 2 above)
Comment (berland) [reply tests/unit_tests/resources/test_ecl_run_new_config.py]: fixed, thx

**Pull request merged**